feat/PointOfSalePage

XartCircleX opened the pull request:
smn, no agregue feat/ ni en el nombre ni en el commit 🥹

XartCircleX committed:
feat: implement Point of Sale component with product search and cart functionality

XartCircleX committed:
feat: implement Point of Sale component with product search and cart functionality

XartCircleX committed:
fix/ update button label to clarify action for sale products

XartCircleX committed:
fix/ update button label to clarify action for sale products

DonAnchondoBis commented:
<h2>Coverage Report</h2>
<table> <thead> <tr> <th align="center">Status</th> <th align="left">Category</th> <th align="right">Percentage</th> <th align="right">Covered / Total</th> </tr> </thead> <tbody> <tr> <td align="center">🔵</td> <td align="left">Lines</td> <td align="right">15.55%<br/>⬇️ <em>-1.55%</em></td> <td align="right">875 / 5626</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Statements</td> <td align="right">15.55%<br/>⬇️ <em>-1.55%</em></td> <td align="right">875 / 5626</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Functions</td> <td align="right">83.33%<br/>⬆️ <em>+1.19%</em></td> <td align="right">70 / 84</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Branches</td> <td align="right">89.65%<br/>⬆️ <em>+0.05%</em></td> <td align="right">338 / 377</td> </tr> </tbody> </table><details><summary>File Coverage</summary><table> <thead> <tr> <th align="left">File</th> <th align="right">Stmts</th> <th align="right">Branches</th> <th align="right">Functions</th> <th align="right">Lines</th> <th align="left">Uncovered Lines</th> </tr> </thead> <tbody> <tr> <td align="left" colspan="6"><b>Changed Files</b></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/page.jsx">src/app/page.jsx</a></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/page.jsx#L2-L51" class="text-red">2-51</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/Navbar/utils.jsx">src/app/Navbar/utils.jsx</a></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/Navbar/utils.jsx#L2-L70" class="text-red">2-70</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/UI/Images/index.js">src/app/UI/Images/index.js</a></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/UI/Images/index.js#L1-L6" class="text-red">1-6</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/api/ticket/route.js">src/app/api/ticket/route.js</a></td> <td align="right">90.24%<br/>⬇️ <em>-3.27%</em></td> <td align="right">80.76%<br/>⬇️ <em>-2.57%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">90.24%<br/>⬇️ <em>-3.27%</em></td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/api/ticket/route.js#L26-L30" class="text-red">26-30</a>, <a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/api/ticket/route.js#L69-L75" class="text-red">69-75</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/landingPageComponents/Dashboard/Dashboard.jsx">src/app/landingPageComponents/Dashboard/Dashboard.jsx</a></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/landingPageComponents/Dashboard/Dashboard.jsx#L3-L128" class="text-red">3-128</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/pointOfSale/page.jsx">src/app/pointOfSale/page.jsx</a></td> <td align="right">0%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/d43d35793ea355d44c0be4dd60e26e1e1087d2ae/src/app/pointOfSale/page.jsx#L2-L658" class="text-red">2-658</a></td> </tr> </tbody> </table></details>
<em>Generated in workflow <a href=https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/actions/runs/15947835680>#372</a> for commit <a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/commit/d43d35793ea355d44c0be4dd60e26e1e1087d2ae">d43d357</a> by the <a href="https://github.com/davelosert/vitest-coverage-report-action">Vitest Coverage Report Action</a></em>

<!-- vitest-coverage-report-marker-root -->

TheSixDecadence commented on src/app/landingPageComponents/PointOfSale/PointOfSale.jsx:
```diff
@@ -1,9 +1,349 @@
+  gridTemplateColumns: '1fr 5fr',
+  justifyContent: 'space-between',
+  padding: '1rem',
+  borderTop: `1px solid ${theme => theme.palette.primary.main}`,
```
Nice code Bryan! The only correction that I can make you is the syntax for borderTop. You're using a function inside a template string, which doesn’t work.
To fix it, you should return an object with a function that receives theme, like this:
```
const ContainerButtons = styled('div')(({ theme }) => ({
  borderTop: `1px solid ${theme.palette.primary.main}`,
}));
```

XartCircleX replied:
oks

TheSixDecadence reviewed (changes requested):
The code is clean, well structured, and easy to follow. The only change I asked you to make is minor. Great work!

XartCircleX committed:
fix/correct borderTop property syntax in ContainerButtons styling

XartCircleX committed:
fix/correct borderTop property syntax in ContainerButtons styling

XartCircleX committed:
fix/correct ContainerButtons styled component to use theme for borderTop property

XartCircleX committed:
fix/correct ContainerButtons styled component to use theme for borderTop property

Salvador511 committed:
fix: change the path of the pointOfSale, to improve the routing

Salvador511 committed:
fix: change the path of the pointOfSale, to improve the routing

Salvador511 committed:
fix: update link path for Point Of Sale to ensure correct routing

Salvador511 committed:
fix: update link path for Point Of Sale to ensure correct routing

Alejandra160 committed:
feat: add NotFound component with 404 error handling (#31)

Co-authored-by: Salvador511 <salvadororozcogon511@gmail.com>

Alejandra160 committed:
feat: improve error handling in API responses and set default props for tables (#32)

Co-authored-by: Salvador511 <salvadororozcogon511@gmail.com>

KamenDeku committed:
Fix/bug fixes (#33)

* feat(acquisitions): rename purchasing to acquisitions and implement new components

- Updated Navbar to reflect the change from "Purchasing" to "Acquisitions".
- Added LotsTable component for displaying lots with pagination and search functionality.
- Introduced ModalAddLot component for adding new lots with form validation.
- Created ModalAddSupplier component for adding new suppliers with form validation.
- Implemented SuppliersTable component for displaying suppliers with pagination and search functionality.
- Developed main acquisitions page to manage lots and suppliers, including modals for adding new entries.
- Added utility functions for form validation and initial values for lots and suppliers.
- Updated Dashboard to link to the new acquisitions page.

* feat: add EmptyState component and integrate it into reports and acquisitions pages; update Navbar with Home and Logout options; fix label in ModalAddSupplier

* fix: lock

---------

Co-authored-by: Salvador511 <salvadororozcogon511@gmail.com>

RaulMun committed:
feat/Cloudinary4Products  (#26)

* feat: add image field to Product model and implement image upload in POST /api/product

* feat: update Product to include an image with cloudinary

* feat: rename image field in Product model to imageUrl for clarity

* chore: remove image field from Product model and update related tests and routes for image handling

* fix: adjust Content-Type header handling in apiFetch for FormData support

---------

Co-authored-by: Salvador511 <salvadororozcogon511@gmail.com>

RaulMun committed:
feat: integrate product fetching and enhance styling for Point Of Sale component

RaulMun committed:
feat: integrate product fetching and enhance styling for Point Of Sale component

RaulMun committed:
feat: update Point Of Sale component to use Next.js Image component

RaulMun committed:
feat: update Point Of Sale component to use Next.js Image component

RaulMun committed:
feat: add stock validation for ticket creation in POST method

RaulMun committed:
feat: add stock validation for ticket creation in POST method

RaulMun committed:
feat: ticket test reworked and verified

RaulMun committed:
feat: ticket test reworked and verified

RaulMun committed:
feat: add search type selector for product filtering in Point Of Sale and default image when the item does not have one

RaulMun committed:
feat: add search type selector for product filtering in Point Of Sale and default image when the item does not have one

XartCircleX committed:
feat: enhanced ui, cart fixes and dialog component

XartCircleX committed:
feat: enhanced ui, cart fixes and dialog component

XartCircleX committed:
feat: sell button disable, better responsive, search icon in search bar, added delete product stack and only one.

XartCircleX committed:
Merge branch 'PointOfSalePage' of https://github.com/DonAnchondoBis/AbarrotesDonAnchondo into PointOfSalePage,
